app/services/data_set_documentation.rb

```diff
@@ -12,4 +12,19 @@ def self.for(klass)
     used_common_columns = common_columns.slice(*spec['common_columns'])
     used_common_columns.merge(spec['custom_columns'] || {})
   end
+
```

Comment:
I'm in two minds with this class and would appreciate some feedback. There are a few different options

- use instance variables, which means that we only need to load the yml files once. But this class is used in a few different places and currently has a neat concise notation in this format 

```<% DataSetDocumentation.for(DataAPI::TADExport).each do |column_name, column| %>
  <%= render DataSetAttributeComponent.new(column_name: column_name, column: column) %>
<% end %>```

- stick with using "static" methods but use something like `private_class_method` or `class << self` so that we can pull out some of the repeated logic into private methods. This helps to dry out the class, but it would still load the files twice.

It's only in our export test that it would load the files twice. For the rest of the uses it only uses the `self.for` method and so loads them once. So I'm leaning towards the second option https://stackoverflow.com/questions/29737435/ruby-class-with-static-method-calling-a-private-method

Reply:
I've just pushed those changes so you can see what they look like

Reply:
:+1: I think loading twice is fine.